### ISSUE TITLE
Keyboard navigation not working in UI menus

### ISSUE THREAD

**Issue by thienautran** [danny-avila/LibreChat#3177]
#### Bug 
When using the avatar settings menu and the ai model switcher menu, keyboard arrows and enter are not working as expected.

in the settings menu (accessible by clicking the avatar on the bottom left), arrows can navigate up and down correctly, however pressing enter on a highlighted option will not trigger the action. The only way to trigger the action is to click on it with the mouse. 

In the ai model switcher arrow keys don’t move the focus to a model in the list and there’s no real way to select a model with arrows / or tab and enter. 

#### Expected behaviour
What should ideally happen in both cases is that users are able to select things with the keyboard and activate them with enter. 

For the settings menu, moving up and down works, enter just needs to trigger the item that’s selected.  

For the model switcher there are two ways that this could be fixed:
1. Approach 1 - arrows similar to google search box: When the model switcher is open, default focus should be on the search bar, using arrow keys can select and highlight a model. The user can navigate up and down using the arrows to select a model. Pressing enter would activate the selected model, typing would enter text inside the search bar. If search returns no results, enter should do nothing. This is similar to how Google search works
2. Approach 2 - tab to shift system focus between list and search bar: When the picker is open, default focus is on the search box.  Pressing tab will transport focus from search box to list of models, from here the user could use arrows to select a model and press enter to confirm the selection and close the picker.

### Steps to Reproduce

Settings menu:
1. Click the avatar on the bottom left to bring up the menu actions
2. Use arrow keys to highlight an option
3. Press enter to try to activate an item
4. Observe menu has dismissed, however action has not been activated

model switcher:
1. Click the button to select a ai model to bring up the picker
2. Press arrows or tab to try to move system focus 
3. Observe how system focus doesn’t always highlight a model
4. Observe how enter does not activate a selection

### What browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
> Settings menu:

This is now working as intended, as is no longer reproducible

> model switcher:

The issues you highlighted can no longer be observer

@thienautran We've been tackling some of these issue piecemeal, it would be helpful if you could help us track what's missing still in scoped issues

Please visit the a11y issues to view and maybe add anything you see missing, or even review the closed issues to see they are indeed resolved.

https://github.com/danny-avila/LibreChat/issues?q=is%3Aissue+is%3Aopen+label%3Aa11y
